encodings/dict/src/array.rs

```diff
@@ -97,9 +97,10 @@ impl ArrayCanonicalImpl for DictArray {
         match self.dtype() {
             // NOTE: Utf8 and Binary will decompress into VarBinViewArray, which requires a full
             // decompression to construct the views child array.
-            // For this case, it is *always* faster to decompress the values first and then create
-            // copies of the view pointers.
-            DType::Utf8(_) | DType::Binary(_) => {
+            // For this case, it is faster to decompress the values first and then create
+            // copies of the view pointers. If this array has more values than codes, we do not
+            // canonicalize the entire values child, and fallback to take
+            DType::Utf8(_) | DType::Binary(_) if self.values().len() < self.codes().len() => {
```

Comment:
Did you run any benchmarks, It might still be better to canonicalize if there are a few more values than codes

Reply:
for compute I did a while back, as soon as values grow larger then codes It was beneficial to canonicalise on compute: https://github.com/spiraldb/vortex/blob/develop/encodings/dict/benches/dict_compare.rs#L71-L114